onnx2pytorch/convert/model.py

```diff
@@ -59,30 +69,85 @@ def __init__(
         self.batch_dim = batch_dim
         self.experimental = experimental
         self.debug = debug
+        self.enable_pruning = enable_pruning
+
+        self.input_names = get_inputs_names(onnx_model.graph)
+        self.output_names = get_outputs_names(onnx_model.graph)
+        opset_version = onnx_model.opset_import[0].version
+
+        # Create mapping from node (identified by first output) to submodule
         self.mapping = {}
-        for op_id, op_name, op in convert_operations(onnx_model, batch_dim):
+        for op_id, op_name, op in convert_operations(
+            onnx_model.graph,
+            opset_version,
+            batch_dim,
+            enable_pruning,
+        ):
             setattr(self, op_name, op)
+            if isinstance(op, Loop) and debug:
+                raise NotImplementedError("debug-mode with Loop node not implemented.")
             self.mapping[op_id] = op_name
 
-        self.init_parameters = InitParameters(
-            {tensor.name: tensor for tensor in self.onnx_model.graph.initializer}
-        )
+        # Store initializers as buffers
+        for tensor in self.onnx_model.graph.initializer:
+            buffer_name = get_buffer_name(tensor.name)
+            self.register_buffer(
+                buffer_name,
+                torch.from_numpy(np.copy(numpy_helper.to_array(tensor))),
+            )
 
-        self.input_names = get_inputs_names(onnx_model)
+        # Compute activation dependencies, mapping each node to its dependents
+        self.needed_by = defaultdict(set)
+        for node in self.onnx_model.graph.node:
```

Comment:
Could you move this to a separate method? Call it: compute_activation_dependencies.
Reason is to keep functions and methods small and quickly understandable.

Reply:
Done!